### PR TITLE
Feature like&bookmark

### PR DESCRIPTION
**Pull request by bhtibrewal**
- the UI for a single post page is WIP.
- added comments are only visible on single post page
- to go to the single post page click on 'n Comment'

pages to check:
- src/components/user_section/UserSection.jsx
- src/components/card/PostCard.jsx
- src/components/comment_section/CommentSection.jsx

### CODE REVIEW

File: src/components/card/PostCard.jsx [hunk 1]
```diff
@@ -1,93 +1,100 @@
...
-          />
-        </div>
-        <div>
-          <p className="user-name">
-            {firstName} {lastName}
-          </p>
-          <p className="user-handle">{username}</p>
-        </div>
-      </div>
+      <UserSection user={{ username, profile_pic, firstName, lastName }} />
```
Comment (Naman-Saxena1) [inline src/components/card/PostCard.jsx:68]: This is fine but if you are just passing to other component and again destructuring in UserSection.jsx Line 5.
I think you can just directly pass object without extra destructuring at Line 68

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
I tried liking and bookmarking post, it redirects to login page but when I try login as guest, I am unable to login, so maybe you can cross check that.

1. Create Post and Post component should have same width to get alignment.
2. I suggest you disable textarea horizontal resize using resize: vertical; property on it.
Capture1

File: src/pages/auth/sign_in/SignIn.jsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ export const SignIn = () => {
   const navigate = useNavigate();
   const location = useLocation();
   const from = location?.state?.from?.pathName || "/";
+  console.log(location);
```
Comment (SinchanaMS) [inline src/pages/auth/sign_in/SignIn.jsx:14]: Remove logs after checks.

File: src/services/posts/addComment.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        const res = await axios.post(`/api/posts/comments/${postId}`, { comment });
+        console.log(res);
+        if (res.status === 201) {
+            postsStateDispatch({ type: "UPDATE_POSTS", payload: res.data.posts });
+            showToast({ title: 'comment posted', type: 'success' });
+        }
+    }
+    catch (e) {
+        console.log(e)
+        showToast({ title: e?.response?.errors, type: 'error' });
```
Comment (SinchanaMS) [inline src/services/posts/addComment.js:14]: Please check if this is working. I can see only the toast, and no message when I try adding the comment.
image

File: src/services/posts/likePost.js [hunk 4]
```diff
@@ -1,11 +1,15 @@
 import axios from "axios"
 
-export const likePost = async ({postId}) => {
+export const likePost = async ({ postId, postsStateDispatch, showToast }) => {
     try {
         const res = await axios.post(`/api/posts/like/${postId}`);
-        console.log(res);
+        console.log(res)
```
Comment (SinchanaMS) [inline src/services/posts/likePost.js:6]: Please remove the log here too.

**Review by SinchanaMS — APPROVED** [review]
The code looks good.
However, there are some issues I found in the UI, please look into them.

1. Guest login is not working.
2. The error toast when the user is not logged in is not showing the error message.
3. The post text-area is resizing beyond the container horizontally. Either enable only vertical resize or add a limit to the horizontal resize property to not overflow.

**Review by prankurpandeyy — APPROVED** [review]

**Comment by prankurpandeyy** [comment]
all code looks good to me , however login as guest isnt working

**Pull request merged**